### DISCUSSION TITLE
Testing the apis

### DISCUSSION THREAD

**Discussion by sarthakrepos** [appium/appium#20677] [Q&A]
I cloned the code.
Able to build and run.
But I am not getting how to get session id to test the apis.

**Comment by portyu9** [comment]
Hi @sarthakrepos 

You get the session ID by **creating an Appium session first**.

Most Appium API endpoints are session-scoped, so the normal flow is:

```text
POST /session
        ↓
receive sessionId
        ↓
/session/<sessionId>/...
```

For example, with an Android device and the UiAutomator2 driver installed, a W3C session request can look like:

```bash
curl -X POST http://127.0.0.1:4723/session \
  -H 'Content-Type: application/json' \
  -d '{
    "capabilities": {
      "alwaysMatch": {
        "platformName": "Android",
        "appium:automationName": "UiAutomator2"
      },
      "firstMatch": [{}]
    }
  }'
```

The response contains the created session ID. Use that value in APIs such as:

```text
GET /session/<sessionId>/screenshot
GET /session/<sessionId>/source
POST /session/<sessionId>/element
```

If you create the session through Appium Inspector or a WebDriver client instead, that session already has an ID; you can also see it in the Appium server log.

One version-related detail: Appium 1 used `/wd/hub` as the default base path. Appium 2+ uses `/` by default, so current URLs normally start with:

```text
http://127.0.0.1:4723/session
```

rather than:

```text
http://127.0.0.1:4723/wd/hub/session
```

Current Appium also has `GET /appium/sessions` for discovering active sessions, but that endpoint requires the `session_discovery` insecure feature to be explicitly enabled. Normally it is simpler to keep the session ID returned by `POST /session`.

Please mark this answer as accepted if it helped, thank you.
